pandas/_libs/tslibs/conversion.pyx

```diff
@@ -32,6 +50,31 @@ cdef int64_t DAY_NS = 86400000000000LL
 
 UTC = pytz.UTC
 
+# ----------------------------------------------------------------------
+# Misc Helpers
+
+cdef inline bint _is_timestamp(datetime obj):
+    # datetime instance but not datetime type --> Timestamp
+    return not PyDateTime_CheckExact(obj)
+
+
+# TODO: We can type the input as a np.datetime64 right?
+# and the output as an int64_t?
+cdef inline _get_datetime64_nanos(object val):
+    cdef:
+        pandas_datetimestruct dts
+        PANDAS_DATETIMEUNIT unit
+        npy_datetime ival
+
+    unit = get_datetime64_unit(val)
+    ival = get_datetime64_value(val)
+
+    if unit != PANDAS_FR_ns:
+        pandas_datetime_to_datetimestruct(ival, unit, &dts)
+        check_dts_bounds(&dts)
+        return dtstruct_to_dt64(&dts)
+    else:
+        return ival
```

Comment:
de-privatize

note I am not only suggesting these things for general cleanliness, but also because these prevent accidental imports; IOW things will immediately break because of the new name unless they are fixed